addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -134,6 +134,12 @@ const Wysiwyg = Widget.extend({
                     if (!$field.hasClass('o_editable_date_field_format_changed')) {
                         $linkedFieldNodes.text($field.data('oe-original-with-format'));
                         $linkedFieldNodes.addClass('o_editable_date_field_format_changed');
+                        const originalDate = $field.data('oe-original');
+                        $(`.oe_hide_on_date_edit[data-oe-original='${originalDate}']`)
+                            .addClass('d-none');
```

Comment:
This is not correct, it will also alter other record's date if they have the same date, you need to run that selector on `$linkedFieldNodes`
```suggestion
                        const originalDate = $field.data('oe-original');
                        $linkedFieldNodes.filter('.oe_hide_on_date_edit').addClass('d-none');
```
otherwise you end up like https://drive.google.com/file/d/1KHoze2p4HysS-eNC_COdOlP4teWs4JBn/view (Designed fair was modified when I didnt clicked on it)